### ISSUE TITLE
Would like a lint to handle unused tearoffs (or similar)

### ISSUE THREAD

**Issue by eseidelGoogle** [dart-lang/linter#1105]
https://twitter.com/pavel_ismailov/status/1004641813548892161

Is the report, which came in via twitter.

```dart
foo.someFunction;
```
Is the typo which they wish the analyzer would have warned them about.

**Comment by pq** [comment]
@bwilkerson, @scheglov : any reason this shouldn't be flagged as unnecessary by the analyzer proper?

**Comment by MichaelRFairhurst** [comment]
This is handled by unnecessary_statements, which it looks like is not enabled in the project:


flutter/analysis_options.yaml:    
```
# - unnecessary_statements # not yet tested
```

unnecesary_statements gives `list.clear;` intsead of `list.clear()` as an example:

```                                                                
**BAD:**                                                                                                                                              
list.clear;
```

This appears to have test coverage:

```
class MyClass {
  int foo() => null;
}

main() {
  new MyClass().foo; // LINT
}
```

I'm making a PR now to turn this on in flutter.

**Comment by MichaelRFairhurst** [comment]
Not sure if I'm doing this right, but, PR here: https://github.com/flutter/flutter/pull/20099

**Comment by pq** [comment]
Awesome.  As I mentioned there, if it lands in flutter, you'll want to add it to https://github.com/flutter/plugins/blob/master/analysis_options.yaml as well.
